### ISSUE TITLE
Login Webpage Response Doesn't Trigger Login within Fusion360

### ISSUE THREAD

**Issue by warasilapm** [cryinkfly/Autodesk-Fusion-360-for-Linux#418]
**Describe the bug**
Fusion is able to launch the web browser page to login, however, the response that should be handled by AdskIdentityManager.exe doesn't seem to do anything. The .launch exists and Firefox picks up on it, nothing seems to happen. I _do_ see an "invisible" asdkidentity manager window when I use the alt-tab style window switching on my system, but whenever I select it the main Fusion 360 window appears instead. It's worth noting that until I kill AdskIdentityManager.exe  using the system monitor tool the Fusion360 window doesn't have a way to interact with it; I guess it's blocked on the child window that isn't drawing?
![image](https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/assets/37279515/4278006c-c667-4ad2-90c8-ac0192e2bf7d)
![image](https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/assets/37279515/f45a05ee-a88e-476c-872e-4c25a48dec72)

**To Reproduce**
1. Install using install.sh.
2. Attempt to log in.
3. Log in within the native browser.
4. Wait forever twiddling thumbs like a neanderthal.

**Expected behavior**
The login page response is handled by F360/AdskIdentityManager.exe.

**Screenshots**

**Desktop (please complete the following information):**
- Version number of Autodesk Fusion 360: No idea, it's whatever comes using install.sh at the moment.
- Graphics card model + Graphics card driver (Version number): Ryzen 7 Pro 5850U Integrated Graphics
- Operating system (OpenSUSE Leap 15.3, Fedora 34, ...): Pop!_OS 22.04 LTS (basically Ubuntu 22.04)
- Desktop environment (XFCE 4.16, Gnome 40, ...): Gnome 42.9
- DXVK- or OpenGL-Mode: DXVK
- Wine: 9.8(staging)

**Additional context**
I had tried using bottles a few months ago but that suffered from the login issues resolved in #381 and I figured I'd try the install script to avoid my confusion with bottles.

I can only imagine this is a system specific quirk that I need to address but I don't even know where to start on this one.

**Comment by cooldoggy** [comment]
Try to inspect the .desktop file for AdskIdentityManager.exe, on my PC the path was corrupted and once corrected to point at the executable properly it worked.

**Comment by MatzeS** [comment]
Based on @cooldoggy 's comment I fixed my setup, by adjusting the ID in the path of the desktop file.
I also replaced some backslashes with regular slashes.

While debugging your setup, consider checking `journalctl`. `krunner` prints some error log if launching the application fails.
Also not forget to `update-desktop-database`.

**Comment by warasilapm** [comment]
Yep. The `webdeploy\production\[key]\` path had the wrong `[key]` value. I wonder why that is? Something that is missed when updating from one version to another?

**Comment by warasilapm** [comment]
Should I start a new issue specifically for this path issue to capture it as an item to resolve long term?

**Comment by jrgilman** [comment]
I just wanted to +1 this, that the same fix worked for me, for some reason the uuid/key value that @warasilapm is referring to was completely wrong on mine, fixing it to the actual key made Fusion login right away! Thanks @cooldoggy.

**Comment by BedoAkgun** [comment]
changing the fusion 360 folder the .desktop file was pointing at (webdeploy\production\random letters) didnt help, fusion still freezes after clicking sign in on the app. I also did "sudo update-desktop-database"

**Comment by warasilapm** [comment]
> changing the fusion 360 folder the .desktop file was pointing at (webdeploy\production\random letters) didnt help, fusion still freezes after clicking sign in on the app. I also did "sudo update-desktop-database"

That seems like a unique problem.

**Comment by gidoBOSSftw5731** [comment]
> Based on @cooldoggy 's comment I fixed my setup, by adjusting the ID in the path of the desktop file. I also replaced some backslashes with regular slashes.
> 
> While debugging your setup, consider checking `journalctl`. `krunner` prints some error log if launching the application fails. Also not forget to `update-desktop-database` after editing the `.desktop` file.

I'll admit that I'm not super familiar with the backend of this project, however I don't see anything in the desktop file that refers to the webdeploy/production/id path? Is there a different .desktop?
https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/blob/main/files/builds/stable-branch/.desktop/Autodesk%20Fusion%20360.desktop

**Comment by Shourya-maker-lab** [comment]
Hey i am on fedora 40 and i have downloaded fusion from the installer package  yesterday i am i able to launch it and sign in the browser but its not redirecting and i have wine Staglin and i have tried the script which was provided to resolve this issue but noting happens i am tried of it pls help

**Comment by Shourya-maker-lab** [comment]
And what is the .desktop thing everybody is talking about ?

**Comment by warasilapm** [comment]
@gidoBOSSftw5731 @Shourya-maker-lab 

We're referring to the adskidmgr-opener.desktop file used to open the web callback on sign in. This is hardcoded into the install script. See below.

https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/blob/af88102a797e28d65dea5bef8418e79539d7b8ed/files/builds/stable-branch/bin/install.sh#L409-L417

Not sure if there is a good way to change this to reflect the random value correctly.

**Comment by gidoBOSSftw5731** [comment]
> @gidoBOSSftw5731 @Shourya-maker-lab
> 
> We're referring to the adskidmgr-opener.desktop file used to open the web callback on sign in. This is hardcoded into the install script. See below.
> 
> https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/blob/af88102a797e28d65dea5bef8418e79539d7b8ed/files/builds/stable-branch/bin/install.sh#L409-L417
> 
> Not sure if there is a good way to change this to use the specific webpack value correctly.

Ah, thanks for the clarification. I figured out my laptop hadn't been updated in a while (yay libc6 upgrades) so it was having a ton of issues, I've now gotten some progress working with bottles (after updating that too.) I added reactions to posts on #393 for the posts I had to follow to get it to work.

**Comment by Reutertu3** [comment]
Which link is supposed to open even and in what browser? Upon launching Fusion there is an error involving QtWebEngineProcess.exe, which closes itself. After that the Fusion Login screen appears, but clicking on Sign-In doesn't do anything.

**Comment by Shourya-maker-lab** [comment]
to get till the login issue use firefox browser after sign in in auto desk webiste you need to inspect the page which will have a button called go to product and you need to copy the link associated with the button.

**Comment by FireNWater** [comment]
Same issue here.  Is there anyone who can explain the fix a bit more clearly?

**Comment by PMarinov1994** [comment]
I update the path to AdskIdentityManager.exe with the correct ID just like @warasilapm said, but I still cannot log in (the fusion360 window is frozen)

**Comment by FireNWater** [comment]
Same here. .

**Comment by ho-ho-ho** [comment]
@PMarinov1994 I had the same problem ... changing all the `\` to `/` in the Exec line of the desktop file made it work.

**Comment by major-mayer** [comment]
I also have the same issue and changing all `\` to `/` results in a new error message (when I launch Firefox from console):
```
wine: failed to open "C:/Program Files/Autodesk/webdeploy/production/99249ee497b13684a43f5bacd5f1f09974049c6b/Autodesk Identity Manager/AdskIdentityManager.exe": c0000135
```

Autodesk doesn't show any reaction and is stuck in the welcome screen.
Any idea?

**Comment by Cyborgnetical** [comment]
having same issue in Pop_OS! The solution above worked.

For any of you confused  (or I might just be dumb cuz it took me a while to figure this out) here are my steps that worked.
 1. open up file explorer, go to your Home > .local > share > applications > adskidmgr-opener.desktop (with a text editor)
 2. open up another file explorer,  again your Home > C_drive > Program Files > Autodesk > webdeploy > production
     check which folder has Autodesk Identity Manager, copy the name of the folder that has it. 
3. in your text editor change the long mess of charaters (99249ee497b13684a43f5bacd5f1f09974049c6b) to what you copied.
tada :) now try log in again

**Comment by thenicnet** [comment]
https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/issues/418#issuecomment-2132533054

THANK YOU!
I was trying to decipher these one line responses and what people meant. 

This worked perfectly and now I'm rocking on Pop_OS!

**Comment by FireNWater** [comment]
Got it working here, too on Manjaro.  Thanks for the less cryptic explanation!!  Now to get my 3D mouse working. .

**Comment by Shourya-maker-lab** [comment]
will try today on a fresh fedora install hopefully it works . on fedora 40 .
@cryinkfly if this works we can add that to the code i guess ? .

**Comment by Shourya-maker-lab** [comment]
@FireNWater  that is like a everytime you open the fusion 360 or just once in a while ?

**Comment by cryinkfly** [comment]
@Shourya-maker-lab :+1:

**Comment by FireNWater** [comment]
> @FireNWater  that is like a everytime you open the fusion 360 or just once in a while ?
> 
> 

Yes, the error seems to trigger every time Fusion opens a new window. The data panel doesn't open and eventually Fusion freezes and I have to kill it from the OS. 

I'd be happy to collect log files, just let me know which ones you want.

**Comment by Aus-gez** [comment]
I don't have anything to contribute for the QTWebEngine issues... Sorry.

But I have been working on a patch to add a small function to identify the variable location of the identity manager, specifically the alphanumeric folder name. This will hopefully mean we no longer need to hard code the identity manager location (which also tripped me up).

I've not tested it yet as I ran out of time tonight, but if anyone wants to have a go at testing it and let me know how you go, here are the changes I made:
https://github.com/Aus-gez/Autodesk-Fusion-360-for-Linux/commit/2a72b14ce63306516b31931208f4614f6105eff6

Hopefully I didn't put the function in a place that will run before the identity manager exists on the file system. :grimacing:

**Comment by wociscz** [comment]
Confirm working workaround. I can login now, but gazillion of `QTWebEngineProcess.exe` errors poping out.

**Comment by Shourya-maker-lab** [comment]
i can conform that the workaround is not working on fedora 40

**Comment by Shourya-maker-lab** [comment]
and ubuntu 24.04 Lts

**Comment by Shourya-maker-lab** [comment]
@zakutin  pls elobrate on the fedora 40 
![image](https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/assets/89152020/29b372c7-f108-4484-8683-617ac24c176e)

**Comment by Shourya-maker-lab** [comment]
please tell the steps from start to end please. thankyou <3

**Comment by wociscz** [comment]
confirm: the Qt bug fix with editing the .dll works. Only problem I'm having is I cant export/save the model as stl. After clicking export (save as or 3d print) nothing happens.

**Comment by warasilapm** [comment]
> confirm: the Qt bug fix with editing the .dll works. Only problem I'm having is I cant export/save the model as stl. After clicking export (save as or 3d print) nothing happens - utilities -> make -> works.

This may be an Fusion bug in general; I've had a couple launches on windows have similar issues. The first time I saw this, it crashed Fusion outright. A different time, it hung for several minutes before allowing me to export the stl. Could be unrelated, though.

**Comment by Aus-gez** [comment]
The original issue of not being redirected back to Fusion 360 after logging in should now be resolved.

This means the folder that the identity manager is installed in is now correctly identified instead of being hard coded.

**Comment by ZeXx86** [comment]
I can confirm the bug on at least two different computers with Ubuntu 22.04 with a fresh Fusion 360 install.
None can log in. Web browser is not opened at all.

**Comment by ZeXx86** [comment]
Is there a way I can find why I am geting crash? On other computer Firefox is crashing too when Login page should open.
Is there a chance to find link it is opening?

**Comment by uatec** [comment]
If Chrome is crashing, change your default browser.

**Comment by ZeXx86** [comment]
Behavior is identical on two different computers with different web browsers and fresh Fusion installation.
Either with Chrome and Firefox I can see browser will Crash. However both browsers are running normally when started manually. It does not matter if browser is running at the event of crash or not running. It will never open anything.

If I will find what link is to be opened I can find reason for crash. Maybe there is some strange character or something like this.

**Comment by Aus-gez** [comment]
@ZeXx86 , sorry to hear you're still having issues logging in.
I haven't run in to an issue with browsers crashing trying to load the URL for logging in.

For the record I did my testing on PopOS 22.04. Be good to know if anyone else is having issues or successes with the patch.

Can you please do 2 bits of troubleshooting for me? This will check the bits my patch affected.

Please post the content of this file: ~/.local/share/applications/adskidmgr-opener.desktop

...and the output of the 'find' command after running these lines:
```sh
cd ~/.fusion360/wineprefixes/
find . -name 'AdskIdentityManager.exe'
```

**Comment by ZeXx86** [comment]
> @ZeXx86 , sorry to hear you're still having issues logging in. I haven't run in to an issue with browsers crashing trying to load the URL for logging in.
> 
> For the record I did my testing on PopOS 22.04. Be good to know if anyone else is having issues or successes with the patch.
> 
> Can you please do 2 bits of troubleshooting for me? This will check the bits my patch affected.
> 
> Please post the content of this file: ~/.local/share/applications/adskidmgr-opener.desktop
> 
> ...and the output of the 'find' command after running these lines:
> 
> ```shell
> cd ~/.fusion360/wineprefixes/
> find . -name 'AdskIdentityManager.exe'
> ```

```
tomas@ubuntu:~$ cd ~/.fusion360/wineprefixes/
find . -name 'AdskIdentityManager.exe'
./default/drive_c/Program Files/Autodesk/webdeploy/production/0b67cdf9b5e1511700618e51757ca1535bfaa4da/Autodesk Identity Manager/AdskIdentityManager.exe
tomas@ubuntu:~/.fusion360/wineprefixes$  cat  ~/.local/share/applications/adskidmgr-opener.desktop
[Desktop Entry]
Type=Application
Name=adskidmgr Scheme Handler
Exec=env WINEPREFIX="/home/tomas/.fusion360/wineprefixes/default" wine "C:\Program Files\Autodesk\webdeploy\production\0b67cdf9b5e1511700618e51757ca1535bfaa4da\Autodesk Identity Manager\AdskIdentityManager.exe" %u
StartupNotify=false
Terminal=false
MimeType=x-scheme-handler/adskidmgr;
tomas@ubuntu:~/.fusion360/wineprefixes$ 

```

**Comment by Aus-gez** [comment]
@ZeXx86 - thanks. Those look correct to me, which means your issue could be a new kind of login issue. ...which is what you suspected in your first mention of this issue. 

Possibly something to do with KDE and how these links are handled before they get passed to the browser? 

Which version of wine are you using?  Can you please show the output of `wine --version`?

You said you tried it on a couple of machines. Were they both KDE based environments too?
If so, are you in a position to try a gnome based environment in a virtual (or actual) machine eg Ubuntu or PopOS?

**Comment by ZeXx86** [comment]
> @ZeXx86 - thanks. Those look correct to me, which means your issue could be a new kind of login issue. ...which is what you suspected in your first mention of this issue.
> 
> Possibly something to do with KDE and how these links are handled before they get passed to the browser?
> 
> Which version of wine are you using? Can you please show the output of `wine --version`?
> 
> You said you tried it on a couple of machines. Were they both KDE based environments too? If so, are you in a position to try a gnome based environment in a virtual (or actual) machine eg Ubuntu or PopOS?

Thanks for your reply. Wine is: wine-9.10 (Staging)

Yes, both machines are originally running KDE. However on one of these I have also Gnome. Behavior there is same. I am just getting crash report popup window from Gnome. Both machines are running different CPU (AMD/Intel) and GPU (iGPU AMD/NVidia RTX) so it is not related to hardware I think.

I can try virtual machine with any OS.

**Comment by ZeXx86** [comment]
I tried to reproduce the bug in VirtualBox with a very fresh Ubuntu 22.04.4 LTS installation without any modification. I can confirm the problem is identical to my. After clicking at the Sign In nothing will happen - Fusion 360 login screen will just freeze or waiting for something - and nothing will happen anymore. Same as with KDE on my machines.

**Comment by Aus-gez** [comment]
I'll download Ubuntu 22.04.4 LTS and give it a try too. It'll probably be a couple of days before I can get back to you though - got some packed days. 

Hopefully in the meantime others will share their experience and we'll track down the cause of this issue.

**Comment by ZeXx86** [comment]
> I'll download Ubuntu 22.04.4 LTS and give it a try too. It'll probably be a couple of days before I can get back to you though - got some packed days.
> 
> Hopefully in the meantime others will share their experience and we'll track down the cause of this issue.

Thank you very much. It would be very helpful since I use Fusion quite often and I am running Linux everywhere.

Please note with Ubuntu Wayland could be used. With Wayland running it might not start at all even under Xwayland. So to be more specific I set X11 session. Depending on where you run it, Wayland could add another problems..

**Comment by Aus-gez** [comment]
@ZeXx86 I managed to spin up the VM using Ubuntu 22.04.4 and it did crash. I've still got more testing to do, but I my theory is that the Snap package of these browsers is the cause of the issues. 

I won't be able to try this until tomorrow, but next steps are to see if this issue happens with the Deb version of Firefox: https://www.omgubuntu.co.uk/2022/04/how-to-install-firefox-deb-apt-ubuntu-22-04

If this works, then that might be enough for you to go on with for now. Meanwhile we'll be a step closer to properly solving it.

Let me know the results if you try this in your Ubuntu VM.

**Comment by ZeXx86** [comment]
I can confirm it works! Thank you very much!
Interestingly I had Google Chrome installed from DEB previously and it does not worked when it was set as default browser.

**Comment by warasilapm** [comment]
> I can confirm it works! Thank you very much! Interestingly I had Google Chrome installed from DEB previously and it does not worked when it was set as default browser.



> I can confirm it works! Thank you very much! Interestingly I had Google Chrome installed from DEB previously and it does not worked when it was set as default browser.



> I can confirm it works! Thank you very much! Interestingly I had Google Chrome installed from DEB previously and it does not worked when it was set as default browser.

I, too, did not have any success with chromium or edge on Pop!_OS 22.04 LTS. I only had success with Firefox.

**Comment by ZeXx86** [comment]
I confirm it works only with Firefox. I tested also Konqueror - this worked, but it is unable to redirect back to Fusion 360. So for some reason only Firefox (not from Snap) is working.

**Comment by b14ckw1d0w** [comment]
I can't manage to login even after checking the folder and .desktop file match and modifying the Qt6WebEngineCore.dll file manually and trying the one that is modified above. I have tried to get it to login with Firefox and Brave but both request to open the handler app but nothing proceeds and Fusion is still stuck not logged in.

I am running on Arch with a fresh never before used installation of Fusion 360.

How can I see what is going wrong as I am receiving no output or errors from anything after patching Qt6WebEngineCore.dll.

**Comment by Aus-gez** [comment]
I've got no clue what is causing snap-based browsers to crash.

I tried running the address in the crash report and that actually worked... which really confused me. I'm reporting it here in case someone else has an "ah ha!" moment.

In this case I was using the snap package for Chromium.

The command that is run looks like this:
```
/snap/bin/chromium https://developer.api.autodesk.com/authentication/v2/authorize?client_id=W4OzUBRrCUtlVRIgARbv5rRIKMHe7dXp&response_type=code&response_mode=query&scope=user-profile%3Aread%20user%3Aread%20user%3Awrite%20viewables%3Aread%20data%3Aread%20data%3Awrite%20data%3Acreate%20data%3Asearch%20bucket%3Acreate%20bucket%3Aread%20bucket%3Aupdate%20bucket%3Adelete%20code%3Aall%20account%3Aread%20account%3Awrite%20openid%20device_sso&redirect_uri=https%3A%2F%2Faccounts.autodesk.com%2Fidmgr%2Fcallback&code_challenge_method=S256&code_challenge=BGBEMGCrK32JxllMZO4O6lLuCxyDC2gyUe6Q2L_bwhQ&state=VzRPelVCUnJDVXRsVlJJZ0FSYnY1clJJS01IZTdkWHAuQkdCRU1HQ3JLMzJKeGxsTVpPNE82bEx1Q3h5REMyZ3lVZTZRMkxfYndoUS51c2VyLXByb2ZpbGUlM0FyZWFkJTIwdXNlciUzQXJlYWQlMjB1c2VyJTNBd3JpdGUlMjB2aWV3YWJsZXMlM0FyZWFkJTIwZGF0YSUzQXJlYWQlMjBkYXRhJTNBd3JpdGUlMjBkYXRhJTNBY3JlYXRlJTIwZGF0YSUzQXNlYXJjaCUyMGJ1Y2tldCUzQWNyZWF0ZSUyMGJ1Y2tldCUzQXJlYWQlMjBidWNrZXQlM0F1cGRhdGUlMjBidWNrZXQlM0FkZWxldGUlMjBjb2RlJTNBYWxsJTIwYWNjb3VudCUzQXJlYWQlMjBhY2NvdW50JTNBd3JpdGUlMjBvcGVuaWQlMjBkZXZpY2Vfc3NvLnVzZV9hbHRlcm5hdGl2ZV9wcm9kdWN0X3BhdGg=

```
Weirdly, if run the full command, it'll fail to redirect properly but it won't cause a crash. 

Anyone got any ideas why snap-based browsers are causing crash reports when being called for the login attempt?

**Comment by ZeXx86** [comment]
I think it has nothing to do with Snap. The same is happening with Chrome from DEB.
It is either system library problem or the apps have hardcoded buffer for certain amount of characters without any checking causing segfault.

**Comment by figa12** [comment]
The solution proposed by @zakutin worked for me. Thank you!

**Comment by kb-elmo** [comment]
This still doesn't work for me.

After clicking on sign in in the Fusion starter the Autodesk login page opens in the browser (tried multiple ones) but after that the Fusion starter just freezes indefinitely and never actually starts the program.

The path in the desktop file for the AdskIdentityManager is correct (checked it multiple times over and over) and even running the adskidmgr manually with the callback code from the login page doesn't do anything.

`WINEPREFIX="/home/<user>/.fusion360/wineprefixes/default" wine "C:\Program Files\Autodesk\webdeploy\production\b738a7f16475c27414ba957f7f0b2ee5b34e4e6d\Autodesk Identity Manager\AdskIdentityManager.exe" "adskidmgr:/login?code=XXXXXXXXXXX"`

Unfortunately there also don't seem to be any errors in the Fusion log visible.
The program UI just gets stuck somewhere and never updates, buttons become unclickable etc.

**Comment by kb-elmo** [comment]
@zakutin yeah I can run it (using i3wm with rofi-drun to run desktop file) and it doesn't cause any error.
Also no error when starting it from the terminal.

Also already tried running it with the absolute unix path instead of the Wine Windows path already (found that in another comment on some issue here) but it doesn't work either unfortunately.

Same exact result every time. Browser opens, I can login but then the Fusion splashscreen just freezes.

**Comment by paradoxx42** [comment]
I am somehow still unable to log in. I am using Firefox from the arch repo and used the install.sh script to install fusion.
The command
`❯ find . -name "AdskIdentityManager.exe"`
returns:
`
./.fusion360/wineprefixes/default/drive_c/Program Files/Autodesk/webdeploy/production/b738a7f16475c27414ba957f7f0b2ee5b34e4e6d/Autodesk Identity Manager/AdskIdentityManager.exe`

My adskidmgr-opener.desktop contains the following:

```
[Desktop Entry]
Type=Application
Name=adskidmgr Scheme Handler
Exec=env WINEPREFIX="/home/josua/.fusion360/wineprefixes/default" wine "/home/josua/.fusion360/wineprefixes/default/drive_c:/Program Files/Autodesk/webdeploy/production/b738a7f16475c27414ba957f7f0b2ee5b34e4e6d/Autodesk Identity Manager/AdskIdentityManager.exe"%u
StartupNotify=false
MimeType=x-scheme-handler/adskidmgr;
```

Any idea what could be going wrong here? I already rebooted as well...

**Comment by wrennnnnn** [comment]
> I am somehow still unable to log in. I am using Firefox from the arch repo and used the install.sh script to install fusion. The command `❯ find . -name "AdskIdentityManager.exe"` returns: ` ./.fusion360/wineprefixes/default/drive_c/Program Files/Autodesk/webdeploy/production/b738a7f16475c27414ba957f7f0b2ee5b34e4e6d/Autodesk Identity Manager/AdskIdentityManager.exe`
> 
> My adskidmgr-opener.desktop contains the following:
> 
> ```
> [Desktop Entry]
> Type=Application
> Name=adskidmgr Scheme Handler
> Exec=env WINEPREFIX="/home/josua/.fusion360/wineprefixes/default" wine "/home/josua/.fusion360/wineprefixes/default/drive_c:/Program Files/Autodesk/webdeploy/production/b738a7f16475c27414ba957f7f0b2ee5b34e4e6d/Autodesk Identity Manager/AdskIdentityManager.exe"%u
> StartupNotify=false
> MimeType=x-scheme-handler/adskidmgr;
> ```
> 
> Any idea what could be going wrong here? I already rebooted as well...

Your desktop exec after wine is wrong, it should be a windows style path, like `Exec=env WINEPREFIX="/home/josua/.fusion360/wineprefixes/default" wine "C:\Program Files\Autodesk\webdeploy\production\b738a7f16475c27414ba957f7f0b2ee5b34e4e6d\Autodesk Identity Manager\AdskIdentityManager.exe" %u`

**Comment by paradoxx42** [comment]
I have read multiple times that I should be putting an absolute path, so I’m kinda puzzled at this point

Josua Schmidt
+49 162 4916811
Nürnberg, Deutschland
________________________________
Von: Wren ***@***.***>
Gesendet: Tuesday, June 25, 2024 4:11:24 AM
An: cryinkfly/Autodesk-Fusion-360-for-Linux ***@***.***>
Cc: paradoxx42 ***@***.***>; Comment ***@***.***>
Betreff: Re: [cryinkfly/Autodesk-Fusion-360-for-Linux] Login Webpage Response Doesn't Trigger Login within Fusion360 (Issue #418)


I am somehow still unable to log in. I am using Firefox from the arch repo and used the install.sh script to install fusion. The command ❯ find . -name "AdskIdentityManager.exe" returns: ./.fusion360/wineprefixes/default/drive_c/Program Files/Autodesk/webdeploy/production/b738a7f16475c27414ba957f7f0b2ee5b34e4e6d/Autodesk Identity Manager/AdskIdentityManager.exe

My adskidmgr-opener.desktop contains the following:

[Desktop Entry]
Type=Application
Name=adskidmgr Scheme Handler
Exec=env WINEPREFIX="/home/josua/.fusion360/wineprefixes/default" wine "/home/josua/.fusion360/wineprefixes/default/drive_c:/Program Files/Autodesk/webdeploy/production/b738a7f16475c27414ba957f7f0b2ee5b34e4e6d/Autodesk Identity Manager/AdskIdentityManager.exe"%u
StartupNotify=false
MimeType=x-scheme-handler/adskidmgr;


Any idea what could be going wrong here? I already rebooted as well...

Your desktop exec after wine is wrong, it should be a windows style path, like Exec=env WINEPREFIX="/home/josua/.fusion360/wineprefixes/default" wine "C:\Program Files\Autodesk\webdeploy\production\b738a7f16475c27414ba957f7f0b2ee5b34e4e6d\Autodesk Identity Manager\AdskIdentityManager.exe" %u

**Comment by GruenSein** [comment]
Is there any particular trick when using Bottles? I have changed the prefix and the path in the .desktop-file (tried absolute paths, windows-style paths etc.). The browser reliably opens but it never triggers the login so I am still stuck at the login window.

**Comment by cryinkfly** [comment]
If any here that can helps me here: https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/issues/421#issuecomment-2269554243

**Comment by cryinkfly** [comment]
So i have added the patched Qt6WebEngineCore.dll to the installer: https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/tree/main?tab=readme-ov-file#%EF%B8%8F-getting-started and https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/commit/db93e9dacddc26923ad023fa8cedc43a733cf929

**Comment by guillouGabrielEtude** [comment]
Hello, I'm reopening this post because the DLL is out of date. This causes an error at the beginning of fusion.

I realized this because I installed fusion via bottle as I am on an immutable linux distribution and the automatic patch of the dll was not applied.

so i followed this post to patch manually and i realized that it wasn't possible because the line to modify was no longer `03d81230: 01cc` but `03d81230: 0000`.

I therefore have 2 problems to list for this subject:
- the absence of automatic patching under bottles.
- and the outdated dll

I'd like to do the fix but I have no idea how to fix a hexadecimal file or a dll.

excuse me for my bad english. i'm using a translator because i'm not very good and thanks to those who will look into the problem.

Sincerely Gabriel Guillou
